libsolidity/interface/CompilerStack.cpp

```diff
@@ -741,16 +741,20 @@ void CompilerStack::compileContract(
 	// 16-bit big endian length
 	cborEncodedMetadata += toCompactBigEndian(cborEncodedMetadata.size(), 2);
 
-	// Run optimiser and compile the contract.
-	compiler->compileContract(_contract, _compiledContracts, cborEncodedMetadata);
-
 	try
 	{
-		compiledContract.object = compiler->assembledObject();
+		// Run optimiser and compile the contract.
+		compiler->compileContract(_contract, _compiledContracts, cborEncodedMetadata);
```

Comment:
Actually `compileContract` is calling `optimise` and that is the only one which can (should) output `OptimizerException`.